pkgsettings/pkgsettings.py

```diff
@@ -26,6 +31,36 @@ def as_dict(self):
             result.update(item.as_dict())
         return result
 
+    def children(self):
+        """
+        Tries to return a generator of all settings objects in the chain, recursively.
+        This might not yield all settings objects, if they include
+        other settings objects not by using the children() call.
+        :return: generator of settings objects.
+        """
+        for settings in self._chain:
+            yield settings
+
+        for child in self._chain:
+            children = getattr(child, 'children', None)
+            if callable(children):
+                for settings in children():
+                    yield settings
```

Comment:
By using a generator + building a list (in`_has_duplicates()`), we can detect recursion in the data structure even though we use recursion to walk through the data structure.